frontend_tests/node_tests/stream_data.js

```diff
@@ -128,6 +135,58 @@ test("basics", () => {
     assert.equal(stream_data.slug_to_name("99whatever"), "99whatever");
 });
 
+test("get_subscribed_streams_of_user", () => {
+    const denmark = {
+        subscribed: true,
+        color: "blue",
+        name: "Denmark",
+        stream_id: 1,
+        is_muted: true,
+        invite_only: true,
+        history_public_to_subscribers: true,
+    };
+    const social = {
+        color: "red",
+        name: "social",
+        stream_id: 2,
+        is_muted: false,
+        invite_only: false,
+        history_public_to_subscribers: false,
+        stream_post_policy: stream_data.stream_post_policy_values.admins.code,
+    };
+    const test = {
+        color: "yellow",
+        name: "test",
+        stream_id: 3,
+        is_muted: true,
+        invite_only: true,
+    };
+    const subs = [denmark, social, test];
+    for (const sub of subs) {
+        stream_data.add_sub(sub);
+    }
+
+    peer_data.set_subscribers(denmark.stream_id, [me.user_id, test_user.user_id]);
+    peer_data.set_subscribers(social.stream_id, [test_user.user_id]);
+    peer_data.set_subscribers(test.stream_id, [test_user.user_id]);
+
+    blueslip.expect(
+        "warn",
+        "We got a is_user_subscribed call for a non-existent or inaccessible stream.",
+    );
+    // test_user is subscribed to all three streams, but current user(me)
+    // gets only two because of subscriber visibility policy of stream:
+    // #denmark: current user is subscribed to it so he can see its subscribers.
+    // #social: current user is not subscribed to it but it is not invite only
+    //          and current user is not guest so he can see its subscribers.
+    // #test: current user is not subscribed to but it is invite only, so
+    //        he can not see if test_user is subscribed to it.
```

Comment:
I don't understand this test setup.  If the current user has the stream in their `stream_data`, then they can see it.

Reply:
This test setup ensures that we show only those streams whose subscribers can be seen by current users. Like a normal user cannot see subscriber list of a invite only stream (as appears from `is_user_subscribed` function) so if target user is subscribed to such a stream then we do not show it in to someone who does have access to its subscriber list.

I have updated this comment as well as commit message for clarity.

Reply:
Right, but the reason they can't see subscribers of such a stream is that no subscriber data will be present in `stream_data`...